### DISCUSSION TITLE
포트란 무엇일까요?

### DISCUSSION THREAD

**Discussion by 3Juhwan** [2024-woowacourse-study/level-interview#231] [Network]
.

**Comment by alstn113** [comment]
포트(Port)는 네트워크 통신에서 데이터가 특정 프로세스 또는 서비스로 전달되도록 하는 논리적 경로입니다. 
인터넷 프로토콜(IP)을 사용하여 데이터가 컴퓨터 사이에서 이동할 때, IP 주소는 데이터가 전달될 장치를 식별하고 포트 번호는 데이터를 수신할 특정 프로세스를 식별합니다. 포트는 0번부터 65535번까지 존재합니다. 
ex) IP가 아파트의 단지라면 PORT는 몇동 몇호

포트 번호는 특정 프로토콜과 함께 사용됩니다. 예를 들어, TCP와 UDP가 있으며, 이들은 각각 신뢰성 있는 연결 지향 통신과 빠른 비연결 지향 통신을 제공합니다. 같은 포트 번호라도 프로토콜이 다르면 다른 역할을 할 수 있습니다. 

포트 사용의 예로는 다음과 같은 것들이 있습니다.

- **웹 서버**: 웹 서버는 HTTP 요청을 받기 위해 80번 포트를 사용하고, HTTPS 요청을 받기 위해 443번 포트를 사용합니다.
- **SSH 서버**: 원격으로 서버에 접속하기 위해 22번 포트를 사용합니다.
- **데이터베이스 서버**: MySQL 서버는 기본적으로 3306번 포트를 사용합니다.

**Comment by 3Juhwan** [comment]
포트는 네트워크를 통해 데이터를 송수신할 때 특정 프로세스나 서비스와 통신하기 위해 사용되는 논리적인 채널입니다. 포트는 16비트 숫자로 표현되며, TCP/UDP 프로토콜에서 사용되는 포트 번호는 0부터 65535까지의 범위를 가집니다. 

포트 번호는 3가지로 분류됩니다. 
- Well-Known Ports(0 - 1023): 운영체제에 의해 사용되고 있음
- Registered Ports(1024 - 49151): 애플리케이션에서 이 범위 내의 포트를 사용하여 통신
- Dynamic ports(49152 - 65535): 용도가 정해지지 않음

**Comment by J-I-H-O** [comment]
**<정의>**

포트(Port)는 네트워크 서비스나 특정 프로세스를 식별하는 논리 단위입니다. 
IP주소가 통신할 장치를 찾을 때 필요한 주소라면, Port 번호는 장치 안에서 통신할 프로세스를 식별할 때 사용됩니다.
 TCP/UDP와 같은 전송 계층 프로토콜에서 사용됩니다.

**<분류>**
- 0 ~ 1023 : Well-known port, 잘 알려진 포트
    - 주로 시스템 서비스나 주요 프로토콜에 할당된 포트
- 1024 ~ 49151 : Registered port, 등록된 포트
    - 특정 서비스가 사용하도록 등록된 포트 (예: MySQL → 3306)
- 49152 ~ 65535 : Dynamic port, 동적 포트
    - 주로 클라이언트 측에서 일시적인 통신을 위해 사용되는 포트
    - 클라이언트 애플리케이션이 서버에 연결할 때 임의 할당되어 사용되며, 세션이 종료되면 다시 반환

**<자주 사용되는 포트 번호>**

![image](https://github.com/2024-woowacourse-study/level-interview/assets/110461155/2442de9c-ef71-4a4a-b0e3-6024c95470ec)
